department-of-veterans-affairs/caseflow
# ClayS/APPEALS-39107

Resolves [APPEALS-39107](https://jira.devops.va.gov/browse/APPEALS-39107)

# Description
For VHA non-admins, adds a 'Request issue modification' button that leads to the edit issues page, and disables the 'Edit Issues' button.

For VHA admins and non-VHA users, functionality should be the same.

## Acceptance Criteria
- [ ] Creation of new non-admin ‘Request issue modification’ button
- [ ] Non-Admin behavior:
    - [ ] Can view but are unable to select the ‘Edit Issues’ button (grayed out/disabled).
    - [ ] Can request issue modification through ‘Request issue modification’ button.
    - [ ] Description under ‘Decision’ header/section:
        - [ ] Only VHA admins can make edits to Higher-Level Reviews and Supplemental Claims. If you would like to add, remove, or modify an issue within this claim, please request a modification via the button above.
- [ ] Admin behavior:
    - [ ] Are unable to see the ‘Request issue modification’ button.
    - [ ] Description under ‘Decision’ header/section:
        - [ ] Review each issue and assign the appropriate dispositions. (existing functionality)
- [ ] Button will route to ‘Edit Issues’ page where non-admin can make requests.


## Testing Plan

1. Log in as a VHA non-admin user such as VHAPOUSER and navigate to the VHA decision review queue
2. Click a task and verify 
    a. the new button is present and leads to the edit issues page
    b. edit issues button is disabled
    b. the new text is present
3. Log in as a VHA admin user such as VHAADMIN and navigate to the VHA decision review queue
4. Click a task and verify
    a. the new 'Request issue modification' button is not present
    b. the edit issues button is still there and enabled
    c. the original text remains
5. Log in as a non-VHA user like NCA_QUEUE_USER_KENNEDY and repeat step 4



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
|<img width="1136" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/e7f6a917-2880-4d36-b4d8-8bf32e2cef30">|<img width="1116" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/967d229b-f778-47e9-81e1-0f254d7e6751">|

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
